corehq/apps/sms/views.py

```diff
@@ -2046,3 +2047,32 @@ def page_context(self):
                     + _(" failed to fetch templates. Please make sure the gateway is configured properly.")
                 )
         return context
+
+
+class ConnectMessagingUserView(BaseMessagingSectionView):
+    urlname = 'connect_messaging_user'
+    template_name = 'sms/connect_messaging_user.html'
+    page_title = _("Connect Messaging Users")
+
+    @method_decorator(toggles.COMMCARE_CONNECT.required_decorator())
+    @method_decorator(domain_admin_required)
+    def dispatch(self, *args, **kwargs):
+        return super(ConnectMessagingUserView, self).dispatch(*args, **kwargs)
+
+    @property
+    def page_context(self):
+        page_context = super(ConnectMessagingUserView, self).page_context
+        page_context.update({
+            "create_channel_url": reverse("create_channels", args=[self.domain])
+        })
+        return page_context
+
+
+@toggles.COMMCARE_CONNECT.required_decorator()
+@domain_admin_required
+def create_channels(request, domain, *args, **kwargs):
+    user_links = ConnectIDUserLink.objects.filter(domain=domain)
```

Comment:
Why doesn't this need to filter out links that already have a channel id?

Reply:
connectid handles this (it won't create a new channel, just send back the same id), but agree there is no need to make the extra call.